### PR TITLE
Move to portduino GPIO, adding user button support

### PR DESCRIPTION
**Pull request by jp-bennett**
This moves away from using pigpio for hardware support, getting us partway to being able to do away with a raspbian-specific variant. It also adds support for general GPIO like a user button.

Depends on https://github.com/meshtastic/framework-portduino/pull/8

### CODE REVIEW

File: src/gps/GPS.cpp [hunk 1]
```diff
@@ -18,8 +18,8 @@
 #if defined(NRF52840_XXAA) || defined(NRF52833_XXAA) || defined(ARCH_ESP32)
 HardwareSerial *GPS::_serial_gps = &Serial1;
 #elif defined(ARCH_RASPBERRY_PI)
-// need a translation layer to make _serial_gps work with pigpio https://abyz.me.uk/rpi/pigpio/cif.html#serOpen
-HardwareSerial *GPS::_serial_gps = NULL;
+
+HardwareSerial *GPS::_serial_gps = &Serial1;
 #else
```
Comment (caveman99) [inline src/gps/GPS.cpp:23]: duplicate line, can merge into one, no need for the elif

Comment (jp-bennett) [reply src/gps/GPS.cpp:21]: Agreed, fixed.

File: src/platform/portduino/architecture.h [hunk 2]
```diff
@@ -17,3 +17,5 @@
 #ifndef HAS_TELEMETRY
 #define HAS_TELEMETRY 1
 #endif
+#define HAS_WIRE 1
+#define HAS_SCREEN 1
```
Comment (mverch67) [inline src/platform/portduino/architecture.h:21]: Shouln"t it read #ifndef HAS_WIRE ... #endif to be in line with all other definitions in architecture.h and configuration.h

Comment (jp-bennett) [reply src/platform/portduino/architecture.h:21]: The last commit is my in-progress work, hacks included, trying to get a screen to light up.

File: src/ButtonThread.h [hunk 3]
```diff
@@ -45,8 +48,14 @@ class ButtonThread : public concurrency::OSThread
     // callback returns the period for the next callback invocation (or 0 if we should no longer be called)
     ButtonThread() : OSThread("Button")
     {
-#ifdef BUTTON_PIN
+#if defined(ARCH_RASPBERRY_PI) || defined(BUTTON_PIN)
```
Comment (mverch67) [inline src/ButtonThread.h:51]: This line looks superfluous, as both conditions are alread checked below.

**Pull request merged**